src/helpers/url.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * We currently avoid relying on a large library for URL parsing/building to keep bundle size small,
+ * and avoid env-specific dependencies such as node's `url` or `URL` (which is not available in IE11).
+ *
+ * The helpers here do the bare minimum needed to handle use-cases in this library, and might not
+ * handle more complex scenarios with edge-cases.
```

Comment:
I'm not sure I understand how this is better (using a third party lib instead of a Node built-in modules such as `querystring` or `url` which are pretty much global because of shimming). Surely the consumer's bundle will be bigger if they have to bundle both `url`/`querystring` (I suspect most people are already bundling these as they're used almost universally) _and_ this module's dependency on a third part lib `querystringify`. 🤔

If there's a reason not to use `url` specifically, we could at least use the built-in `querystring` module instead of `querystringify`, so there's less chance users will end up bundling multiple dependencies which do the same thing.

Reply:
Alternatively we could just use `URL`/`URLSearchParams` and ask users to polyfill if they need IE11 support.